### PR TITLE
Change objects to Maps

### PR DESCRIPTION
**Pull request by liayoo**
- Updated objects in BlockPool and TransactionPool to Maps
- Related: https://github.com/ainblockchain/ain-blockchain/issues/797

### CODE REVIEW

File: block-pool/index.js [hunk 1]
```diff
@@ -441,8 +449,8 @@ class BlockPool {
       return;
     }
     if (currentBlockInfo.block.number === 0) {
-      this.hashToBlockInfo[currentBlockInfo.block.hash].notarized = true;
-      this.updateLongestNotarizedChains(this.hashToBlockInfo[currentBlockInfo.block.hash]);
+      currentBlockInfo.notarized = true;
+      this.updateLongestNotarizedChains();
```
Comment (platfowner) [inline block-pool/index.js:453]: question: no need to pass an argument?

File: block-pool/index.js [hunk 2]
```diff
@@ -467,29 +474,31 @@ class BlockPool {
         currentBlockInfo.tallied >= totalAtStake * ConsensusConsts.MAJORITY) {
       logger.info(`[${LOG_HEADER}] Notarized block: ${currentBlockInfo.block.hash} ` +
           `(${currentBlockInfo.block.number} / ${currentBlockInfo.block.epoch})`);
-      this.hashToBlockInfo[blockHash].notarized = true;
-      this.updateLongestNotarizedChains(this.hashToBlockInfo[blockHash]);
```
Comment (platfowner) [inline block-pool/index.js:471]: ditto?

File: block-pool/index.js [hunk 3]
```diff
@@ -501,31 +510,29 @@ class BlockPool {
...
             }
           } else {
             this.cleanUpForBlockHash(blockHash);
-            this.numberToBlockSet[blockNumber].delete(blockHash);
+            blockHashSet.delete(blockHash);
           }
         }
-        if (!this.numberToBlockSet[blockNumber].size) {
-          delete this.numberToBlockSet[blockNumber];
+        if (!blockHashSet.size) {
```
Comment (platfowner) [inline block-pool/index.js:527]: minor) what about === 0?

File: tx-pool/index.js [hunk 4]
```diff
@@ -148,10 +148,10 @@ class TransactionPool {
       return null;
     }
 
-    const addrToTxList = JSON.parse(JSON.stringify(this.transactions));
+    const addrToTxList = _.cloneDeep(this.transactions);
```
Comment (platfowner) [inline tx-pool/index.js:151]: 👍

File: tx-pool/index.js [hunk 5]
```diff
@@ -375,37 +373,34 @@ class TransactionPool {
...
       }
     });
-    for (const address in addrToTxSet) {
-      const sizeBefore = this.transactions[address].length;
-      if (this.transactions[address]) {
-        this.transactions[address] = this.transactions[address].filter((tx) => {
-          return !(addrToTxSet[address].has(tx.hash));
-        })
+    for (const [address, txList] of addrToTxSet.entries()) {
+      const sizeBefore = txList ? txList.length : 0;
```
Comment (platfowner) [inline tx-pool/index.js:399]: question: could txList be not a Set?

File: block-pool/index.js (hunk 1 above)
Comment (liayoo) [reply block-pool/index.js:453]: No.. updateLongestNotarizedChains() expects no arguments. I think I updated the function a while ago and forgot to update some of the invocations of it. Thanks for noticing it though!

File: block-pool/index.js (hunk 2 above)
Comment (liayoo) [reply block-pool/index.js:471]: ditto

File: block-pool/index.js (hunk 3 above)
Comment (liayoo) [reply block-pool/index.js]: done~

File: tx-pool/index.js (hunk 5 above)
Comment (liayoo) [reply tx-pool/index.js]: Fixed some bugs in the logic & updated the variable names to make them clearer, thanks!

**Comment by liayoo** [comment]
Addressed all comments, PTAL @platfowner

File: block-pool/index.js (hunk 1 above)
Comment (platfowner) [reply block-pool/index.js:453]: I see. Thanks for the explain.

File: tx-pool/index.js [hunk 6]
```diff
@@ -375,38 +373,36 @@ class TransactionPool {
...
-      const sizeBefore = this.transactions[address].length;
-      if (this.transactions[address]) {
-        this.transactions[address] = this.transactions[address].filter((tx) => {
-          return !(addrToTxSet[address].has(tx.hash));
-        })
+    for (const [address, invalidTxSet] of addrToInvalidTxSet.entries()) {
+      if (this.transactions.has(address)) {
+        const txList = this.transactions.get(address);
+        const sizeBefore = txList.length;
+        this.transactions.set(address, txList.filter((tx) => !invalidTxSet.has(tx.hash)));
```
Comment (platfowner) [inline tx-pool/index.js:402]: 👍

**Review by platfowner — APPROVED** [review]
LGTM
Thanks for the nice refactoring!

**Comment by liayoo** [comment]
Thanks for the review~ Now merging!

**Pull request merged**